tree/tree/src/TChain.cxx

```diff
@@ -1297,6 +1313,10 @@ Int_t TChain::LoadBaskets(Long64_t /*maxmemory*/)
 ///
 Long64_t TChain::LoadTree(Long64_t entry)
 {
+
+   // At exit, if the tree is available, we propagate the IMT setting to it
+   ROOT::Internal::TChain::TTreeIMTSetterRAII treeImtSetterRAAI(fTree, fIMTEnabled);
```

Comment:
Are there more than one update to `fTree` to justify this added complexity? (So far I can only find (original) line 1515.   Alternatively wouldn't a Setter (`void SetLocalTree(TTree *newtree) { fTree = newtree; ... more stuff; };`) be another clearer alternative?

Reply:
My question here is whether we need to propagate the `fImtEnabled` setting at the beginning of `LoadTree` or at the end. As it stands now, the RAII struct does the equivalent of calling the local setter Philippe proposes just at the end of the function + setting it in any case the function throws an exception. Is this the actual behaviour we want? If the function throws do we need to set the data member?

Reply:
This function is not exception safe as it is.  Even if it was, there will be no point is have the setting done as the exception would theoretically would/should leave `fTree` either unchanged or more likely set to nullptr.

Reply:
So IIUC you are saying that we could call `fTree->SetImplicitMT` at the end of `LoadTree` without any extra logic around it